cmd/litefs/main.go

```diff
@@ -131,32 +138,32 @@ func (m *Main) ParseFlags(ctx context.Context, args []string) (err error) {
 	// Only read from explicit path, if specified. Report any error.
 	if *configPath != "" {
 		return ReadConfigFile(&m.Config, *configPath, !*noExpandEnv)
-	} else {
-		// Otherwise attempt to read each config path until we succeed.
-		var configFound bool
-		for _, path := range configSearchPaths() {
-			if path, err = filepath.Abs(path); err != nil {
-				return err
-			}
+	}
 
-			if err := ReadConfigFile(&m.Config, path, !*noExpandEnv); err == nil {
-				fmt.Printf("config file read from %s\n", path)
-				configFound = true
-				break
-			} else if err != nil && !os.IsNotExist(err) {
-				return fmt.Errorf("cannot read config file at %s: %s", path, err)
-			}
+	// Otherwise attempt to read each config path until we succeed.
+	for _, path := range configSearchPaths() {
+		if path, err = filepath.Abs(path); err != nil {
+			return err
 		}
-		if !configFound {
-			return fmt.Errorf("config file not found")
+
+		if err := ReadConfigFile(&m.Config, path, !*noExpandEnv); err == nil {
+			fmt.Printf("config file read from %s\n", path)
+			return nil
+		} else if err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("cannot read config file at %s: %s", path, err)
 		}
 	}
+	return fmt.Errorf("config file not found")
+}
```

Comment:
I separated out the parsing & the validation of the config to make it easier to test.